polymorphia-frontend/hooks/course/useRoadmap/index.tsx

```diff
@@ -1,11 +1,13 @@
 import { useQuery } from "@tanstack/react-query";
 import { RoadmapService } from "@/services/roadmap";
 import { UseRoadmap } from "@/hooks/course/useRoadmap/types";
+import { useUserDetails } from "@/hooks/contexts/useUserContext";
 
 export function useRoadmap(): UseRoadmap {
+  const { courseId } = useUserDetails();
   const { data, isLoading } = useQuery({
```

Comment:
```suggestion
  const { data, isLoading, isError } = useQuery({
```
and at the end: ` return { data, isLoading, isError }`. Related:
- `polymorphia-frontend/hooks/course/useRoadmap/types.ts`:
```ts
export interface UseRoadmap {
  data: StudentGradableEventResponseDTO[] | undefined;
  isLoading: boolean;
  isError: boolean;
}
```
-  `polymorphia-frontend/app/(logged-in)/roadmap/page.tsx`:
```ts
const { data: roadmap, isLoading, isError } = useRoadmap();
// ...
if(isError) {
  return <ErrorComponent message="Nie udało się załadować danych." />;
}
```